### PR TITLE
Re-adjusted error range for bad schema location, need to filter it out duplicates

### LINKED ISSUE #336: Missing schema generates too many warnings
Create a document, referencing an unreachable schema, then each node of that document will get a warning about it:

Screen Shot 2019-03-28 at 5 10 27 PM

Try with:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<invoice xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
 xsi:noNamespaceSchemaLocation="invoice.xsd">
  <date>2017-11-30</date>
  <number>5235</number>
  <products>
    <product />
  </products>
  <payments>
    <payment amount="770.00" method="credit"/>
  </payments>
</invoice>
```

Moreover, the warning range is incorrect, it selects the closing bracket.

### PR DESCRIPTION
**Pull request by NikolasKomonen**
Fixes #336, #343

### CODE REVIEW

File: org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/extensions/diagnostics/AbstractLSPErrorReporter.java [hunk 1]
```diff
@@ -82,9 +85,14 @@ public String reportError(XMLLocator location, String domain, String key, Object
 		if(adjustedRange == null) {
 			return null;
 		}
+
+		Diagnostic d = new Diagnostic(adjustedRange, message, toLSPSeverity(severity), source, key);
+		if(diagnosticSet.contains(d)) {
```
Comment (fbricon) [inline org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/extensions/diagnostics/AbstractLSPErrorReporter.java]:
```java
if (diagnostics.contains(d)) {
```

File: org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/extensions/diagnostics/AbstractLSPErrorReporter.java [hunk 2]
```diff
@@ -82,9 +85,14 @@ public String reportError(XMLLocator location, String domain, String key, Object
 		if(adjustedRange == null) {
 			return null;
 		}
+
+		Diagnostic d = new Diagnostic(adjustedRange, message, toLSPSeverity(severity), source, key);
+		if(diagnosticSet.contains(d)) {
+			return null;
+		}
+		diagnosticSet.add(d);
```
Comment (fbricon) [inline org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/extensions/diagnostics/AbstractLSPErrorReporter.java]: diagnosticSet is not necessary

File: org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/extensions/diagnostics/AbstractLSPErrorReporter.java [hunk 3]
```diff
@@ -39,13 +40,15 @@
 
 	private final DOMDocument xmlDocument;
 	private final List<Diagnostic> diagnostics;
+	private final HashSet<Diagnostic> diagnosticSet;
```
Comment (fbricon) [inline org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/extensions/diagnostics/AbstractLSPErrorReporter.java]: diagnosticSet is not necessary

**Comment by NikolasKomonen** [comment]
@fbricon updated

**Pull request merged**